twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -135,6 +135,15 @@
   private final Map<String, Map<String, String>> environments;
   private final TwillRuntimeSpecification twillRuntimeSpec;
 
+  /**
+   * Final status of this service when it stops.
+   */
+  public enum FinalStatus {
```

Comment:
Sounds more like `StopStatus` or `CompletionStatus`. Also it should be `private` since it is only used in this class.

Also, please move it before all fields declaration instead of having it in between fields.